comp/core/workloadmeta/collectors/internal/ecs/v4parser.go

```diff
@@ -80,7 +81,11 @@ func (c *collector) getTaskWithTagsFromV4Endpoint(ctx context.Context, task v1.T
 		return v1TaskToV4Task(task), errors.New(err)
 	}
 
-	taskWithTags, err := c.metaV3or4(metaURI, "v4").GetTaskWithTags(ctx)
+	taskWithTags, err := v3or4.NewClient(metaURI, "v4", v3or4.WithTryOption(
+		c.metadataRetryInitialInterval,
+		c.metadataRetryMaxElapsedTime,
+		func(d time.Duration) time.Duration { return time.Duration(c.metadataRetryTimeoutFactor) * d }),
+	).GetTaskWithTags(ctx)
```

Comment:
Instead of always re-creating an identical new client object at each `getTaskWithTagsFromV4Endpoint` invocation, would it be possible to have this client created only once at `Collector` creation ?

```suggestion
	taskWithTags, err := c.client.GetTaskWithTags(ctx)
```

Reply:
The client is not identical because the metaURI differs. We retrieve it from each task's container.

https://github.com/DataDog/datadog-agent/blob/b97b3b8333e2b73923eb1f6bdbfde554d5dd7170/comp/core/workloadmeta/collectors/internal/ecs/v4parser.go#L73